### PR TITLE
added pr labels for gitlab and github, and merge parameters for gitlab

### PR DESCRIPTION
**Pull request by epanoff**
It's my first draft by issue [193](https://github.com/baloise/gitopscli/issues/193)

Please check if I my approach correct. And tell me what to do next? 
As I have already said I can't test on GitHub, but my code create labels on my Gitlab repository correctly.

Command like 
```
python3 gitopscli  deploy --file values.yaml --single-commit --username epanov --password glpat-password --git-provider GITLAB --repository-name yaml --organisation epanov --git-provider-url https://gitlab.my-infra.ru  --auto-merge --create-pr   --values "{replicaCount: 10}" --pr-labels "{'test', 'helm'}" 
```

### CODE REVIEW

File: gitopscli/git_api/bitbucket_git_repo_api_adapter.py [hunk 1]
```diff
@@ -108,3 +108,14 @@ def get_pull_request_branch(self, pr_id: int) -> str:
     def __get_default_branch(self) -> str:
         default_branch = self.__bitbucket.get_default_branch(self.__organisation, self.__repository_name)
         return str(default_branch["id"])
+
+    def merge_pull_request_with_parameters(
+        self,
+        pr_id: int,
+        gitlab_merge_parameters: Dict[str, Any],
```
Comment (christiansiegel) [inline gitopscli/git_api/bitbucket_git_repo_api_adapter.py:115]: The more I look at it: Having "gitlab" in the interface just doesn't feel right. We should name that simply `merge_parameters`/`--merge-parameters`. Who knows, maybe this generic object will be useful for another adapter in the future...

File: gitopscli/cliparser.py [hunk 2]
```diff
@@ -301,6 +311,15 @@ def __deduce_empty_git_provider_from_git_provider_url(
     return updated_args
 
 
+def __check_compatibility_args(args: Dict[str, Any], error: Callable[[str], NoReturn]) -> None:
```
Comment (christiansiegel) [inline gitopscli/cliparser.py:314]: Maybe move that check into the individual adapters. That's where we know for sure if someting is supported or not.

File: gitopscli/git_api/gitlab_git_repo_api_adapter.py [hunk 3]
```diff
@@ -97,8 +97,40 @@ def get_pull_request_branch(self, pr_id: int) -> str:
...
         if default_branch is None:
             raise GitOpsException("Default branch does not exist")
         return str(default_branch.name)
+
+    def add_pull_request_label(self, pr_id: int, pr_labels: Dict[str, Any]) -> None:
+        merge_request = self.__project.mergerequests.get(pr_id)
+        merge_request.labels = pr_labels
+        merge_request.save()
+
+    def merge_pull_request_with_parameters(
```
Comment (christiansiegel) [inline gitopscli/git_api/gitlab_git_repo_api_adapter.py:111]: please don't duplicate code :wink:

**Review by christiansiegel — CHANGES_REQUESTED** [review]
Thanks for the PR and sorry for the late review - a bit busy at the moment.

Please don't forget to also update the docs: https://github.com/baloise/gitopscli/blob/master/docs/commands/deploy.md

What is the issue with testing on Github?

File: gitopscli/cliparser.py [hunk 4]
```diff
@@ -112,6 +113,15 @@ def __create_deploy_parser() -> ArgumentParser:
         type=__parse_bool,
         default=False,
     )
+    parser.add_argument(
+        "--pr-labels", help="JSON object pr labels (Gitlab, Github supported)", type=__parse_yaml, default=None
```
[the remark this answers is not in the document]
Comment (epanoff) [reply gitopscli/cliparser.py]: Renamed

File: gitopscli/commands/deploy.py [hunk 5]
```diff
@@ -28,6 +28,8 @@ class Args(GitApiConfig):
         auto_merge: bool
         json: bool
 
+        pr_labels: Any
+        gitlab_merge_parameters: Any
```
[the remark this answers is not in the document]
Comment (epanoff) [reply gitopscli/commands/deploy.py]: Done

File: gitopscli/cliparser.py (hunk 2 above)
Comment (epanoff) [reply gitopscli/cliparser.py]: Just deleted this check. If not supporting - do nothing

File: gitopscli/git_api/gitlab_git_repo_api_adapter.py (hunk 3 above)
Comment (epanoff) [reply gitopscli/git_api/gitlab_git_repo_api_adapter.py]: ok

**Comment by christiansiegel** [comment]
@epanoff is this ready for review? PR is still in draft mode :wink:

**Comment by epanoff** [comment]
Sorry. I've changed mode.

**Review by christiansiegel — APPROVED** [review]
lgtm. Thanks!

Can you please merge the latest master into you PR branch? We changed to Github actions in the meantime. Just so the tests run before merging the PR :wink:

**Comment by christiansiegel** [comment]
Nevermind, ran the tests locally :robot:

**Pull request merged**